bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/SimuComConfig.java

```diff
@@ -62,7 +63,7 @@
 
     // SimuCom extensions can also provide extension to the SimuCom configuration.
     // This map stores the extension configurations.
-    private final HashMap<String, SimuComConfigExtension> simuComConfigExtensions;

```

Comment:
why neccesssary to remove 'final' here ? I would keep it because initialisation is only done in the constructor as far as I understand

Reply:
Unfortunately, the java compiler is not that intelligent. The initialization is done in the doInit() method which is called from the constructors. The problem here stems actually from the dirty SimuCom code, which is doing Eclipse Extension Point lookup in the parent constructor. Therefore, we have to call the apropriate parent constructor before doing the SimuComConfig initialization. Really ugly, but I did not find a way to make it nicer. 

We should really get rid of this SimuComFramework overhead...

Reply:
You are right, I moved the initialization of this particular field to the declaration.